src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,150 @@
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.location.Location;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+
+@Component
+public class StripeParser extends HtmlParser {
+
+    private static final int PAGE_SIZE = 100;
+    private static final String STRIPE_URL = "https://stripe.com";
+    private static final String JOBS_PAGINATION_LIST_XPATH = "//a[text()='Last']";
+    private static final String VACANCY_TITLE_XPATH = "//h1[@class='Copy__title']";
+    private static final String VACANCIES_URLS_XPATH = "//a[@data-analytics-label='jobs_listings_title_link']";
+    private static final String VACANCY_DESCRIPTION_BLOCK_XPATH = "//div[@class='ArticleMarkdown']//*";
+    private static final String VACANCY_CITIES_XPATH = "//span[@class='JobsListings__locationDisplayName']";
+    private static final String VACANCY_JOB_TYPE_XPATH = "//p[text()='Job type']";
+    private static final String VACANCY_TEAM_XPATH = "//p[text()='Team']";
+
+    public StripeParser(WebClient htmlWebClient,
+                        CompanyRepository companyRepository,
+                        LocationProcessor locationProcessor,
+                        @Value("${url.stripe}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Stripe";
+    }
+    @Override
+    public List<Vacancy> getAllVacancies() throws IOException {
+        HtmlPage page = downloadPage(String.format(url, 0));
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>(fillVacancies(
+                page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+        ));
+
+        for (int i = 1; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancies(
+                    page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+            ));
+        }
+
+        return vacancyList;
+    }
+
+    private int getPageCount(HtmlPage page) {
+        String url = ((DomElement) page.getFirstByXPath(JOBS_PAGINATION_LIST_XPATH)).getAttribute("href");
+        return Integer.parseInt(url.replace("?skip=", "")) / PAGE_SIZE;
+    }
+
+    private List<Vacancy> fillVacancies(List<DomNode> urls, List<DomElement> cities) throws IOException {
```

Comment:
This method should accept only one param - HtmlPage. Why? Because these URLs and cities can be derived from the page inside. What is the point of deriving them above TWICE instead of just doing it inside the method once?